### PR TITLE
[New] Add chimee w/ npm auto-update via single package.json 

### LINKED ISSUE #12148: [Request] Add chimee
**Library name:** chimee
**Library description:** a video player aims to bring wonderful experience on browser 
**Git repository url:** https://github.com/Chimeejs/chimee/tree/master/lib
**npm package name or url** (if there is one): https://www.npmjs.com/package/chimee
**License (List them all if it's multiple):** MIT
**Official homepage:** https://chimee.org/

### PR DESCRIPTION
**Pull request by taylorlroberts7**
Pull request for issue: #12148 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/Chimeejs/chimee
 * Official website (optional, not the repository): https://chimee.org/
 * NPM package url (optional): https://www.npmjs.com/package/chimee
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 17
   - Count of stars: 713
   - Count of forks: 38
 * NPM download stats (optional):
   - Downloads in the last day: 33
   - Downloads in the last week: 152
   - Downloads in the last month: 419

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [ ] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 800 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [ ] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/chimee/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  ],
+  "author": "toxic-johann",
+  "license": "MIT",
+  "homepage": "https://chimee.org/",
+  "npmName": "chimee",
+  "npmFileMap": [
+    {
+      "basePath": "lib",
+      "files": [
+        "**/*"
```
Comment (sufuf3) [inline ajax/libs/chimee/package.json]: I thought the files `index.mjs` and `index.js` can't be used in web front-end.
So let's modify the pattern to let bot doesn't get this file.
Let's revise pattern from `**/*` to `index*.+(min.js|browser.js|css|map)`.
It's will helpful to us and developers. 😃 
Thank you.

**Comment by sufuf3** [comment]
@sashberd It's fine
Actually, if you install this version of the lib from npm. (npm install chimee@0.1.0)
You can't find any files that we need to add.
```
chimee/
├── LICENSE
├── README.md
└── package.json
```

What do you think?
Thanks.

**Comment by PeterDaveHello** [comment]
@sufuf3 any reference about this comment https://github.com/cdnjs/cdnjs/pull/12157#discussion_r150389785 ?

For the current auto-update pattern usage in this lib, do we need to include `css` in the filename pattern?

Not sure if @toxic-johann would like to give advice?

Thanks.

**Comment by sufuf3** [comment]
If we use `index*` pattern, we will include `index.mjs` this file which CDNJS not host, maybe.
And I'm sorry for my careless, we need to add index.js, I'll help modify this. (https://github.com/Chimeejs/chimee/blob/master/build/rollup.config.common.js#L5)
I thought it's for back-end, but now I know I was wrong.

Sorry

Thanks

**Comment by toxic-johann** [comment]
Sorry for the late response.

Hmmm, that's my fault. Our name may be not standard. It's totally front-end. @sufuf3 .

In the `lib`, we offer the file, the build is based on this [table](https://github.com/Chimeejs/chimee#explanation-of-different-build).

So what should we put in cdnjs should be only `index.browser.js` and `index.min.js`.

I think it's better to use `/lib/.*\.(min|browser)\.js/`.

If needed, we can change out name too.

Though we would not consider css for now. But  in case we may add it one day, I think we could modify the pattern into this `/lib/.*\.(min|browser)\.(js|css)/`.

**Comment by sufuf3** [comment]
@taylorlroberts7 Thanks for your response.
I modified the pattern to `!(*.dev).+(js|map|css)` exclude `index.dev.js` which be used in development env in v0.1.1, and I think fine to add `index.js` so I still include them.
The current pattern will get `[ 'index.browser.js', 'index.js', 'index.min.js' ]` these files from all the versions. :)

Many thanks.

**Comment by sufuf3** [comment]
@extend1994 I helped modify the pattern, please help review this PR, thanks.

**Comment by mercyg** [comment]
@sufuf3 don't we need to add the version?

**Comment by sufuf3** [comment]
@mercyg There are two method to add the library.
One is adding the lib via single package.json, another is adding the library with assist.
The first method, we don't need to add the version folder with the files & version filed field in package.json.
The second method, we need all of them.

If you need more info about adding the lib, please see https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#e-adding-a-new-library-by-a-single-packagejson

Many thanks.

**Pull request merged**